### PR TITLE
Message selection WIP

### PR DESCRIPTION
**Pull request by vincentbavitz**
Allow for message selection in chat.

TODO:
- [ ] Use correct buttons from styleguide.
- [ ] Connect events to close button, delete button, and forward button

### CODE REVIEW

File: ts/components/session/icon/SessionIconButton.tsx [hunk 1]
```diff
@@ -3,8 +3,18 @@ import classNames from 'classnames';
...
-  public static defaultProps = SessionIcon.defaultProps;
+interface SProps extends Props {
+  onClick: any;
+}
+
+export class SessionIconButton extends React.PureComponent<SProps> {
+  public static readonly extendedDefaults = {
+    onClick: () => null,
+  };
+  public static readonlydefaultProps = {
```
Comment (Mikunj) [inline ts/components/session/icon/SessionIconButton.tsx]: missing a space between `readonly` and `defaultProps`

File: js/models/conversations.js [hunk 2]
```diff
@@ -231,6 +232,47 @@
       this.messageCollection.forEach(m => m.trigger('change'));
     },
 
+    setMessageSelectionBackdrop() {
+      const messageSelected = this.selectedMessages.size > 0;
+
+      if (messageSelected) {
+        $('.messages li, .messages > div').css({
```
Comment (Bilb) [inline js/models/conversations.js]: Why don't add a class with this css instead?

File: stylesheets/_conversation.scss [hunk 3]
```diff
@@ -165,18 +165,6 @@
   display: inline-flex;
 }
 
-.check-box-visible {
-  transition-duration: 200ms;
```
Comment (Bilb) [inline stylesheets/_conversation.scss]: use of $session-transition ?

File: stylesheets/_conversation.scss [hunk 4]
```diff
@@ -165,18 +165,6 @@
...
-  opacity: 0.1;
-  width: 40px;
-}
-
-.check-box-invisible {
-  transition-duration: 200ms;
```
Comment (Bilb) [inline stylesheets/_conversation.scss]: use of $session-transition ?

File: stylesheets/_session.scss [hunk 5]
```diff
@@ -22,41 +22,80 @@
   font-style: italic;
 }
 
+@font-face {
+  font-family: 'Wasa';
```
Comment (Bilb) [inline stylesheets/_session.scss]: Already added to my PR, we will get some conflicts but that's ok

File: ts/components/session/icon/SessionIconButton.tsx (hunk 1 above)
Comment (vincentbavitz) [reply ts/components/session/icon/SessionIconButton.tsx]: Fixed.

File: stylesheets/_session_theme_dark.scss [hunk 6]
```diff
@@ -1,43 +1,49 @@
...
+    border-bottom: none;
   }
 
-  .module-message__metadata__badge {
-    color: $session-color-dark-4;
-  }
-  .module-message__metadata__date--incoming {
-    color: $session-color-dark-4;
-  }
+  .module-message__author,
```
Comment (Bilb) [inline stylesheets/_session_theme_dark.scss]: USe of scss to simplify this? 

```

.module-message {
  &.__author {
....
}
}
```

File: stylesheets/_conversation.scss (hunk 3 above)
Comment (vincentbavitz) [reply stylesheets/_conversation.scss]: This was an omission -- not needed in code anymore.

File: js/views/conversation_view.js [hunk 7]
```diff
@@ -1399,9 +1403,23 @@
     },
 
     deleteSelectedMessages() {
-      const msgArray = Array.from(this.model.selectedMessages);
+      const selected = Array.from(this.model.selectedMessages);
+      const isModerator = this.model.isModerator(this.model.OUR_NUMBER);
+      let isAllOurs = true;
```
Comment (Mikunj) [inline js/views/conversation_view.js]:
```suggestion
      let isAllOurs = selected.every(message => message.attributes.source === message.OUR_NUMBER);
```

File: ts/components/conversation/ConversationHeader.tsx [hunk 8]
```diff
@@ -289,27 +303,61 @@ export class ConversationHeader extends React.Component<Props> {
...
 
     return (
-      <div className="module-conversation-header">
-        {this.renderOptions(triggerId)}
-        {this.renderBackButton()}
-        <div className="module-conversation-header__title-container">
-          <div className="module-conversation-header__title-flex">
-            {this.renderTitle()}
-            {isPrivateGroup ? this.renderMemberCount() : null}
+      <>
```
Comment (Mikunj) [inline ts/components/conversation/ConversationHeader.tsx]: uhm what's this for? If you want a parent component that doesn't render anything then you can just use `<React.Fragment>`

File: js/views/conversation_view.js (hunk 7 above)
Comment (vincentbavitz) [reply js/views/conversation_view.js]: Great idea

File: stylesheets/_session_theme_dark.scss (hunk 6 above)
Comment (vincentbavitz) [reply stylesheets/_session_theme_dark.scss]: Done

**Pull request merged**